service/src/lib.rs

```diff
@@ -131,6 +131,17 @@ impl<Config: Clone + Send + 'static, S: Service<Config>, F: Fn()> ConnectionMana
                 tracing::warn!("Disconnected");
             }
 
+            loop {
+                match shutdown_tx.receiver_count() {
+                    0 => break,
+                    count => {
+                        tracing::error!("Waiting for {count} tasks to shut down...");
+                        tokio::time::sleep(Duration::from_secs(10)).await;
```

Comment:
This seems quite long no? Also should we move into a const or nah?

Reply:
It's a bit long yea but I didn't want to spam log messages too much. Maybe we can use rate limit for that actually